homeassistant/components/home_connect/light.py

```diff
@@ -84,14 +85,58 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Home Connect light."""
+    known_appliance_ids: set[str] = set()
 
-    async_add_entities(
-        [
+    def get_entities_for_appliance(
+        appliance: HomeConnectApplianceData,
+    ) -> list[HomeConnectLight]:
+        """Get a list of entities."""
+        known_appliance_ids.add(appliance.info.ha_id)
+        return [
             HomeConnectLight(entry.runtime_data, appliance, description)
             for description in LIGHTS
-            for appliance in entry.runtime_data.data.values()
             if description.key in appliance.settings
-        ],
+        ]
+
+    async_add_entities(
+        entity
+        for appliance in entry.runtime_data.data.values()
+        for entity in get_entities_for_appliance(appliance)
+    )
+
+    def handle_paired_or_connected_appliance() -> None:
```

Comment:
These callbacks seem to be similar for all platforms. We can probably make common callbacks that we import and use in the platforms. We can use `functools.partial` to store the correct entity creator as an argument with the callback.

Reply:
This can be refactored in a follow up PR. We don't need to do it in this PR.